flake-module.nix

```diff
@@ -148,8 +148,17 @@ in
                 '';
                 default = { };
               };
+              outputs = mkOption {
+                type = types.attrsOf types.raw;
+                description = ''
+                  The flake outputs generated for this project.
+
+                  This is an internal option, not meant to be set by the user.
+                '';
+              };
             };
-          };
+            config = import ./haskell-project.nix (args // { inherit self pkgs; });
```

Comment:
@roberth Is there a better way of doing this (instead of the `//`, which seems like kind of a hack)?

Reply:
Yeah, I think we can use module machinery for this, and we probably want that for those little modules with haskell-flake overrides anyway.
 - Above, use `projectSubmodule = submoduleWith { specialArgs = { inherit pkgs self; }; modules = [ ..... ] }`. `specialArgs` should be fine here, as we don't think we want modules to affect those parameters (not 100% sure about `pkgs`, but that can be improved later if necessary)
 - Use `imports = [ ./haskell-project.nix ];` here